qutip/wigner.py

```diff
@@ -18,14 +20,29 @@
     from scipy.special import sph_harm_y
 except ImportError:
     from scipy.special import sph_harm
-
     def sph_harm_y(n, m, polar, azimuthal):
         # sph_harm is set for removal.
         # Same function, but changed parameter order
         return sph_harm(m, n, azimuthal, polar)
 
-import qutip
-from qutip import Qobj, ket2dm, jmat
+# Direct imports from core modules to avoid circular imports
+from .core.qobj import Qobj
+from .core.states import coherent
+from .core.operators import displace, jmat
+from .core.tensor import tensor
+
+# Parity operator implementation (named 'parity' to match existing calls)
+def parity(N):
+    """Create a parity operator for N-dimensional Hilbert space.
+    
+    The parity operator is P = sum_n (-1)^n |n><n|
+    """
+    from numpy import diag, arange
+    # Create diagonal matrix with (-1)^n entries
+    data = (-1) ** arange(N)
+    return Qobj(diag(data))
```

Comment:
We have a function to create diagonal operators:
```suggestion
    return qdiags(data)
```
It will need to be imported from `operators`.

Reply:
Applied the qdiags suggestion. The parity operator now uses QuTiP's diagonal operator API as recommended. Functionality verified and working correctly.